src/scheduler/server_info.cpp

```diff
@@ -1132,89 +1131,86 @@ free_resource(schd_resource *resp)
+// Init function
+void server_info::init_server_info ()
+{
+
+	has_soft_limit = 0;
+	has_hard_limit = 0;
+	has_user_limit = 0;
+	has_grp_limit = 0;
+	has_proj_limit = 0;
+	has_all_limit = 0;
+	has_mult_express = 0;
+	has_multi_vnode = 0;
+	has_prime_queue = 0;
+	has_nonprime_queue = 0;
+	has_nodes_assoc_queue = 0;
+	has_ded_queue = 0;
+	has_runjob_hook = 0;
+	node_group_enable = 0;
+	eligible_time_enable = 0;
+	provision_enable = 0;
+	power_provisioning = 0;
+	use_hard_duration = 0;
+	pset_metadata_stale = 0;
+	num_parts = 0;
+	name = empty_str;
+	res = NULL;
+	queues = {};
+	queue_list = NULL;
+	jobs = NULL;
+	all_resresv = NULL;
+	calendar = NULL;
+	running_jobs = NULL;
+	exiting_jobs = NULL;
+	nodes = NULL;
+	unassoc_nodes = NULL;
+	resvs = NULL;
+	nodepart = NULL;
+	allpart = NULL;
+	hostsets = NULL;
+	nodesigs = NULL;
+	qrun_job = NULL;
+	policy = NULL;
+	fstree = NULL;
+	equiv_classes = NULL;
+	buckets = NULL;
+	unordered_nodes = NULL;
+	num_nodes = 0;
+	num_resvs = 0;
+	num_hostsets = 0;
+	server_time = 0;
+	job_sort_formula = NULL;
+	init_state_count(&sc);
+	memset(preempt_count, 0, (NUM_PPRIO + 1) * sizeof(int));
+	liminfo = NULL;
 
-	if ((sinfo = new server_info()) == NULL) {
-		log_err(errno, __func__, MEM_ERR_MSG);
-		return NULL;
-	}
+#ifdef NAS
+	/* localmod 034 */
+	share_head = NULL;
+#endif
 
-	sinfo->has_soft_limit = 0;
-	sinfo->has_hard_limit = 0;
-	sinfo->has_user_limit = 0;
-	sinfo->has_grp_limit = 0;
-	sinfo->has_proj_limit = 0;
-	sinfo->has_all_limit = 0;
-	sinfo->has_mult_express = 0;
-	sinfo->has_multi_vnode = 0;
-	sinfo->has_prime_queue = 0;
-	sinfo->has_nonprime_queue = 0;
-	sinfo->has_nodes_assoc_queue = 0;
-	sinfo->has_ded_queue = 0;
-	sinfo->has_runjob_hook = 0;
-	sinfo->node_group_enable = 0;
-	sinfo->eligible_time_enable = 0;
-	sinfo->provision_enable = 0;
-	sinfo->power_provisioning = 0;
-	sinfo->use_hard_duration = 0;
-	sinfo->pset_metadata_stale = 0;
-	sinfo->num_parts = 0;
-	sinfo->name = empty_str;
-	sinfo->res = NULL;
-	sinfo->queues = {};
-	sinfo->queue_list = NULL;
-	sinfo->jobs = NULL;
-	sinfo->all_resresv = NULL;
-	sinfo->calendar = NULL;
-	sinfo->running_jobs = NULL;
-	sinfo->exiting_jobs = NULL;
-	sinfo->nodes = NULL;
-	sinfo->unassoc_nodes = NULL;
-	sinfo->resvs = NULL;
-	sinfo->nodepart = NULL;
-	sinfo->allpart = NULL;
-	sinfo->hostsets = NULL;
-	sinfo->nodesigs = NULL;
-	sinfo->qrun_job = NULL;
-	sinfo->policy = NULL;
-	sinfo->fstree = NULL;
-	sinfo->equiv_classes = NULL;
-	sinfo->buckets = NULL;
-	sinfo->unordered_nodes = NULL;
-	sinfo->num_nodes = 0;
-	sinfo->num_resvs = 0;
-	sinfo->num_hostsets = 0;
-	sinfo->server_time = 0;
-	sinfo->job_sort_formula = NULL;
+}
 
+// Parametrized Constructor
+server_info::server_info(int limallocflag)
+{
+	init_server_info();
 	if ((limallocflag != 0))
-		sinfo->liminfo = lim_alloc_liminfo();
-	init_state_count(&(sinfo->sc));
-	memset(sinfo->preempt_count, 0, (NUM_PPRIO + 1) * sizeof(int));
+		liminfo = lim_alloc_liminfo();
+}
```

Comment:
This shouldn't be needed.  Now that we have a constructor and copy constructor, one passes 1, and the other passes 0.  Look how I did queue_info.

Reply:
I've changed it to initialize it only in the parameterized constructor in later commits. Do you think we don't need parameterized constructor and should always allocate liminfo?

Reply:
So new_server_info() took a parameter to allocate this or not.  When new_server_info() was called in query, it was 1.  When it was called in dup_server_info(), it was 0.  The reason we did that is we didn't want to allocate one in dup just to copy the old one over.  Since these are two different constructors now, you don't need the parameter.